otherlibs/fiber/pool.ml

```diff
@@ -1,17 +1,27 @@
 open Stdune
 open Core
-open Core.O
-
-type mvar =
-  | Done
-  | Task of (unit -> unit t)
 
 type status =
-  | Open
-  | Closed
+  | Open (* new tasks are allowed *)
+  | Closed (* new tasks are forbidden *)
+
+type runner =
+  | Running (* Firing fibers inside the queue *)
+  | Awaiting_resume of
+      unit k (* Ran out of work. Waiting to be resumed once work is added *)
+  | Awaiting_run (* Just created. No [run] has been called *)
+
+(* A pool consumes tasks from a queue in the context where [run] was executed.
 
-type t =
-  { mvar : mvar Mvar.t
+   It's implemented by a simple queue of thunks and a continuation to resume
+   [run] whenever it runs out of work.
+
+   To optimize this further, we can bake in the operation into [effect] in [Core] *)
+
```

Comment:
Is this newline intentional? It seems like the above comment should in fact be attached to the [t] below.

Reply:
Newline removed.